### PR TITLE
Order controller refactoring

### PR DESCRIPTION
**Pull request by evgenii-nikulin**
- refactored `OrderController`
- added unit tests

### CODE REVIEW

File: src/Feature/Checkout/code/Controllers/OrderController.cs [hunk 1]
```diff
@@ -17,14 +17,17 @@ namespace Wooli.Feature.Checkout.Controllers
...
+    using Foundation.Commerce.Models.Checkout;
     using Foundation.Commerce.Repositories;
-    using Foundation.Commerce.Utils;
     using Foundation.Extensions.Extensions;
 
-    [RoutePrefix(Constants.CommerceRoutePrefix + "/order")]
-    public class OrderController : ApiController
+    using Models.Requests;
+
+    public class OrderController : Controller
```
Comment (larionovadaria) [inline src/Feature/Checkout/code/Controllers/OrderController.cs]: Orders

File: src/Feature/Checkout/code/Controllers/OrderController.cs [hunk 2]
```diff
@@ -33,48 +36,48 @@ public OrderController(IOrderRepository orderRepository)
             this.orderRepository = orderRepository;
```
Comment (larionovadaria) [inline src/Feature/Checkout/code/Controllers/OrderController.cs]: assert

File: src/Feature/Checkout/code/Controllers/OrderController.cs [hunk 3]
```diff
@@ -33,48 +36,48 @@ public OrderController(IOrderRepository orderRepository)
...
         }
 
-        [Route("get/{orderId}")]
-        public IHttpActionResult GetOrder(string orderId)
+        [ResponseType(typeof(CartModel))]
+        [HttpGet, ActionName("getOrder")]
```
Comment (larionovadaria) [inline src/Feature/Checkout/code/Controllers/OrderController.cs]:
```
[HttpGet]
[ActionName("order")]
...

[HttpGet]
[ActionName("get")]
```

File: src/Feature/Checkout/code/Models/Requests/GetOrdersRequest.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    {
+        public DateTime? FromDate { get; set; } = null;
+
+        public DateTime? UntilDate { get; set; } = null;
+
+        [Range(0, int.MaxValue)]
+        public int Page { get; set; } = 0;
+
+        [Range(1, int.MaxValue)]
+        public int Count { get; set; } = 5;
```
Comment (larionovadaria) [inline src/Feature/Checkout/code/Models/Requests/GetOrdersRequest.cs]: 5 is a magical number.

File: src/Feature/Checkout/code/Controllers/OrderController.cs [hunk 5]
```diff
@@ -33,48 +36,48 @@ public OrderController(IOrderRepository orderRepository)
...
-
-            if (model.Success)
-            {
-                return this.JsonOk(model.Data);
-            }
+            return this.Execute(() => this.orderRepository.GetOrderDetails(orderId));
+        }
 
-            return this.JsonError(model.Errors.ToArray(), HttpStatusCode.InternalServerError);
+        [ResponseType(typeof(OrderHistoryResultModel))]
```
Comment (larionovadaria) [inline src/Feature/Checkout/code/Controllers/OrderController.cs]: Return value is weird. If I'm asking for orders I should get back several (or one) order entities.
Investigate how it works on FE, and try to understand is it possible to return enumerable of entities.

File: src/Feature/Checkout/code/Controllers/OrderController.cs [hunk 6]
```diff
@@ -33,48 +36,48 @@ public OrderController(IOrderRepository orderRepository)
             this.orderRepository = orderRepository;
         }
 
-        [Route("get/{orderId}")]
-        public IHttpActionResult GetOrder(string orderId)
+        [ResponseType(typeof(CartModel))]
```
Comment (larionovadaria) [inline src/Feature/Checkout/code/Controllers/OrderController.cs]: Why `CartModel` is return value? It should be some `Order`

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/Feature/Checkout/code/Controllers/OrderController.cs (hunk 5 above)
Comment (stanislav-kozorez) [reply src/Feature/Checkout/code/Controllers/OrderController.cs]: I have doubt that `ResponseType` can be used with MVC controller. Did you face any examples of such usage previously?

File: src/Feature/Checkout/tests/Controllers/OrderControllerTests.cs [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+        }
+
+        private readonly OrdersController controller;
+        private readonly IOrderService orderService;
+
+        [Fact]
+        public void GetOrder_IfExceptionIsThrownInService_ShouldReturnErrorResponse()
+        {
+            // arrange
+            const string request = "request";
```
Comment (larionovadaria) [inline src/Feature/Checkout/tests/Controllers/OrderControllerTests.cs]: - use Autofixture for string constants `var orderId = this.fixture.Create<string>()`
- change it in all of your tests

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/Feature/Checkout/tests/Controllers/OrderControllerTests.cs [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+    using Foundation.Extensions.Models;
+
+    using Models.Requests;
+
+    using NSubstitute;
+    using NSubstitute.ExceptionExtensions;
+
+    using Xunit;
+
+    // TODO: Replace models with new one
```
Comment (larionovadaria) [inline src/Feature/Checkout/tests/Controllers/OrderControllerTests.cs]: Did you create a task in tracking system for that?

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/Feature/Checkout/tests/Controllers/OrderControllerTests.cs (hunk 8 above)
Comment (evgenii-nikulin) [reply src/Feature/Checkout/tests/Controllers/OrderControllerTests.cs]: not yet, I will

File: src/Wooli.sln.DotSettings [hunk 9]
```diff
@@ -43,6 +43,10 @@
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.&#xD;
    See the License for the specific language governing permissions and&#xD;
    limitations under the License.</s:String>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpKeepExistingMigration/@EntryIndexedValue">True</s:Boolean>
```
Comment (larionovadaria) [inline src/Wooli.sln.DotSettings]: What is that?

Comment (evgenii-nikulin) [reply src/Wooli.sln.DotSettings]: i don't know. fixed

**Pull request merged**